legacy_promotions/app/decorators/solidus_legacy_promotions/models/spree/order_decorator.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+module SolidusLegacyPromotions
+  module OrderDecorator
+    def self.prepended(base)
+      base.state_machine do
+        before_transition from: :delivery, do: :apply_shipping_promotions
+      end
+    end
+
+    def apply_shipping_promotions(_transition = nil)
+      ::Spree::Config.promotions.shipping_promotion_handler_class.new(self).activate
+      recalculate
+    end
+    deprecate apply_shipping_promotions: :recalculate, deprecator: ::Spree.deprecator
+
+    ::Spree::Order.prepend(self)
```

Comment:
Just a thought: Maybe including the module changes this?

Reply:
(for the reader: "this" referred to the fact that somehow, `apply_shipping_promotions` was now called with an argument). 

This had nothing to do with how the method got into `Spree::Order`. When I removed the deprecation (as the entire gem is slated for deprecation), behaviour returned to the previous behavior.